src/function/matrix/fft.js

```diff
@@ -0,0 +1,114 @@
+import { arraySize } from '../../utils/array.js'
+import { factory } from '../../utils/factory.js'
+import { isMatrix } from '../../utils/is.js'
+import { clone } from '../../utils/object.js'
+
+const name = 'fft'
+const dependencies = [
+  'typed',
+  'matrix',
+  'addScalar',
+  'multiplyScalar',
+  'divideScalar',
+  'exp',
+  'tau',
+  'i'
+]
+
+export const createFft = /* #__PURE__ */ factory(name, dependencies, ({
+  typed,
+  matrix,
+  addScalar,
+  multiplyScalar,
+  divideScalar,
+  exp,
+  tau,
+  i: I
+}) => {
+  /**
+   * Calculate N-dimensional fourier transform
+   *
+   * Syntax:
+   *
+   *     math.fft(arr)
+   *
+   * See Also:
+   *
+   *      ifft
+   *
+   * @param {Array | Matrix} arr    An array or matrix
+   * @return {Array | Matrix}       N-dimensional fourier transformation of the array
+   */
+  return typed(name, {
+    'Array | Matrix': function (arr) {
+      if (isMatrix(arr)) {
+        return matrix(_ndFft(arr.toArray()), arr.storage())
+      } else {
+        return _ndFft(arr)
+      }
+    },
+
+    any: function (x) {
```

Comment:
Is there a use case of supporting `fft(any)`?

Reply:
Yes agree, let's just remove it then👍

Reply:
Done.